### PR TITLE
make adjustments for GHC 8 support

### PR DESCRIPTION
**Pull request by agentm**
* relax cabal upper bound constraints
* use Control.Monad.Catch (catch,finally) where appropriate as per deprecation warning
* fix unused import warnings

Ran "cabal test" against GHC 7.10.3 and GHC 8.0.1 successfully.

I need to compile against binary, time, and deepseq from GHC 8 because our [project](https://github.com/agentm/project-m36) links against GHC for its scripting engine.

The only remaining warnings in GHC8 are loads of redundant constraint warnings (new in GHC 8), but that will be fixed in a subsequent pull request, if it makes sense to fix them.

### CODE REVIEW

File: src/Control/Distributed/Process/ManagedProcess/Internal/Types.hs [hunk 1]
```diff
@@ -80,10 +81,11 @@ type CallId = MonitorRef
 
 newtype CallRef a = CallRef { unCaller :: (Recipient, CallId) }
   deriving (Eq, Show, Typeable, Generic)
-instance Serializable a => Binary (CallRef a) where
-instance NFData a => NFData (CallRef a) where rnf (CallRef x) = rnf x `seq` ()
+--instance Serializable a => Binary (CallRef a) where
```
Comment (Fuuzetsu) [inline src/Control/Distributed/Process/ManagedProcess/Internal/Types.hs:84]: What's the reason for this change?

Comment (agentm) [reply src/Control/Distributed/Process/ManagedProcess/Internal/Types.hs:84]: GHC 8 introduces warnings for unnecessary constraints. I should have left this chunk out of the pull request since it is non-essential, but please be aware that the GHC 8 generates a lot of new warnings for this project.

**Comment by qnikst** [comment]
Thanks for the patch!

Travis CI is failing but so it did for previous versions and master so I'm merging this.

**Comment by hyperthunk** [comment]
The build seems to fail when compiling distributed-process-extras. 

```
src/Control/Concurrent/Utils.hs:36:18: Not in scope: ‘<$>’
src/Control/Concurrent/Utils.hs:36:32: Not in scope: ‘<*>’
src/Control/Concurrent/Utils.hs:36:36: Not in scope: ‘pure’
src/Control/Concurrent/Utils.hs:36:50: Not in scope: ‘<*>’
src/Control/Concurrent/Utils.hs:36:54: Not in scope: ‘pure’
Installed distributed-process-async-0.2.3
cabal: Error: some packages failed to install:
distributed-process-extras-0.2.1.2 failed during the building phase. The
exception was:
ExitFailure 1
```

I'll try and dig in to this, I'm assuming it's an API having changed somewhere.

**Comment by agentm** [comment]
All of those functions are in the [Prelude](http://hackage.haskell.org/package/base-4.9.1.0/docs/Prelude.html) and I don't see any imports hiding them or excluding Prelude.

Also, I just pulled from this master and am unable to reproduce this with 8.0.1.

Are there any errors above this?

**Comment by hyperthunk** [comment]
It seems to be talking to build d-p-extras. I'm gonna look at modifying the extension library builds to emulate distributed-process' since that has changed to use stack and I think these libs should do the same for travis. I'll tweak and submit a pr

**Comment by qnikst** [comment]
Those functions are from older pre-7.10 GHC where where they were not re-exported by Prelude.
```
import Control.Applicative
... other imports
import Prelude
```
should work for all ghc versions.

**Comment by hyperthunk** [comment]
Shows how long I've been awol! Will update in a branch and see if travis is happy...
